libsql-sqlite3/src/vectorIndexInt.h

```diff
@@ -0,0 +1,81 @@
+#ifndef _VECTOR_INDEX_H
+#define _VECTOR_INDEX_H
+
+#include "sqlite3.h"
+#include "vectorInt.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct DiskAnnIndex DiskAnnIndex;
+typedef struct BlobSpot BlobSpot;
+
+/*
+ * Main type which holds all necessary index information and will be passed as a first argument in all index-related operations
+*/
+struct DiskAnnIndex {
+  sqlite3 *db;         /* Database connection */
+  char *zDb;           /* Database name */
+  char *zName;         /* Index name */
+  char *zShadow;       /* Shadow table name */
+  int nFormatVersion;  /* DiskAnn format version */
+  int nDistanceFunc;   /* Distance function */
+  int nBlockSize;      /* Size of the block which stores all data for single node */
+  int nVectorDims;     /* Vector dimensions */
+  int nNodeVectorType; /* Vector type of each node */
+  int nEdgeVectorType; /* Vector type of each node */
+  int nNodeVectorSize; /* Vector size of each node in bytes */
+  int nEdgeVectorSize; /* Vector size of each edge in bytes */
+};
+
+/*
+ * Simple utility class which holds sqlite3_blob handle poiting to the nRowid (undefined if pBlob == NULL)
+ * Caller can re-load BlobSpot with blobSpotLoad(...) method which will reopen blob at new row position
+ * sqlite3_blob_reopen API can be visibly faster than close/open pair since a lot of check can be omitted
+*/
+struct BlobSpot {
+  u64 nRowid;           /* last rowid for which open/reopen was called; undefined if BlobSpot was never opened */
+  sqlite3_blob *pBlob;  /* BLOB handle */
+  u8 *pBuffer;          /* buffer for BLOB data */
+  int nBufferSize;      /* buffer size */
+  int isWritable;       /* blob open mode (readonly or read/write) */
+  int isInitialized;    /* was blob read after creation or not */
```

Comment:
Would it make sense to turn this into flags compressed in a single u8? or at least make both fields u8 instead of int?